### PR TITLE
Allow user-defined labels on pvcs

### LINKED ISSUE #554: [Enhancement][opensearch] Allow user-defined labels on pvcs
**Is your feature request related to a problem? Please describe.**
Our PVCs get backuped based on a policy defined in the PVC label. However it's currently not possible to define additional labels for the PVC (instead the global labels are applied if .Values.persistence.labels.enabled is true). 
**Describe the solution you'd like**
I'd like to define additional labels that are added besides the global labels. 

**Describe alternatives you've considered**
I don't see another way of doing it. 

**Additional context**
-

### PR DESCRIPTION
**Pull request by simonstumpf**
### Description
Add support for user-defined labels on PVCs.
 
### Issues Resolved
#554 
 
### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by simonstumpf** [comment]
accidentally deleted #555, sorry for that

**Comment by peterzhuamazon** [comment]
Hi @simonstumpf sorry we just merged a feature in #586 which bumped the version again.

Would you mind update your version?

Thanks.

**Comment by peterzhuamazon** [comment]
Adding @prudhvigodithi to take a look here.

Thanks.

**Comment by simonstumpf** [comment]
Hi @peterzhuamazon 
Sure, no worries. I've updated the files.

**Review by peterzhuamazon — COMMENTED** [review]
The change seems fine for me.
Adding @prudhvigodithi to take a look on this.

Thanks.

File: charts/opensearch/values.yaml [hunk 1]
```diff
@@ -205,6 +205,8 @@ persistence:
   labels:
     # Add default labels for the volumeClaimTemplate of the StatefulSet
     enabled: false
+    # Add custom labels for the volumeClaimTemplate of the StatefulSet
+    additionalLabels: {}
   # OpenSearch Persistent Volume Storage Class
```
Comment (prudhvigodithi) [inline charts/opensearch/values.yaml:210]: Thanks @simonstumpf there is already `Values.persistence.labels` which is enabled to false by default, what is the expected behaviour here for `additionalLabels` do we have to `Values.persistence.labels.enabled` to true? From code looks to me like we dont have to enable and can still pass the `additionalLabels` to the PVC's.

Comment (peterzhuamazon) [reply charts/opensearch/values.yaml:210]: Hi @simonstumpf would you mind check the comments from @prudhvigodithi ?

Thanks.

Comment (simonstumpf) [reply charts/opensearch/values.yaml:210]: Hi all, 
Sorry I was on a mini vacation, just returned to the office. 

@prudhvigodithi : Sorry I totally ignored this labels.enabled field during my first commit. 
In my point of view the logic should be that regardless of labels.enabled, the additional labels should always be applied. 

I therefore did some changes to the labels logic, could you have another look?

Comment (prudhvigodithi) [reply charts/opensearch/values.yaml:210]: Thanks @simonstumpf, so regardless of the `labels.enabled` (enabled or disabled) if the `persistence.labels.additionalLabels` is passed then the `additionalLabels` will be applied to the PVC's right ? Also again regardless of the `labels.enabled` (enabled or disabled) if the `persistence.labels.additionalLabels` is not passed (no labels) then there wont be any `additionalLabels` applied, am I right here?

Comment (simonstumpf) [reply charts/opensearch/values.yaml:210]: @prudhvigodithi  Yes, you are right

**Comment by prudhvigodithi** [comment]
Great LGTM thanks @simonstumpf, can you please update the bottom section of the changelog file, here is the example PR.
https://github.com/opensearch-project/helm-charts/pull/589/files.

**Comment by simonstumpf** [comment]
> Great LGTM thanks @simonstumpf, can you please update the bottom section of the changelog file, here is the example PR. https://github.com/opensearch-project/helm-charts/pull/589/files.

@prudhvigodithi Added it now, thx

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks I have approved the PR, we can merge once the CI checks pass.
Adding @peterzhuamazon.

**Comment by peterzhuamazon** [comment]
Thanks @simonstumpf would you mind backport the change to 1.x branch?

Thanks.

**Pull request merged**